py/mpz.c

```diff
@@ -1390,6 +1390,24 @@ void mpz_pow3_inpl(mpz_t *dest, const mpz_t *lhs, const mpz_t *rhs, const mpz_t
     mpz_free(n);
 }
 
+#if MICROPY_INT_BIT_LENGTH
+mp_uint_t mpz_bit_length_inpl(mpz_t *n) {
+    if (n->len == 0) {
+        return 0;
+    }
+
+    mpz_t *dest = mpz_clone(n);
```

Comment:
Instead of copying the mpz into a new integer and modifying it, you can just count the digits in the mpz.